app/scripts/emissions-factors-sync.ts

```diff
@@ -0,0 +1,308 @@
+import { PublisherCreationAttributes } from "@/models/Publisher";
+import { MethodologyCreationAttributes } from "@/models/Methodology";
+import { EmissionsFactorCreationAttributes } from "@/models/EmissionsFactor";
+import { DataSourceEmissionsFactorCreationAttributes } from "@/models/DataSourceEmissionsFactor";
+import env from "@next/env";
+import { randomUUID } from "node:crypto";
+import { logger } from "@/services/logger";
+
+interface EmissionsFactorPublisher {
+  name: string;
+  URL: string;
+  publisher_id: string;
+}
+
+interface EmissionsFactorDataSource {
+  datasource_name: string;
+  dataset_name: string;
+  URL: string;
+  publisher_id: string;
+  datasource_id: string;
+}
+
+interface EmissionsFactorMethodology {
+  methodology_id: string;
+  methodology: string;
+  methodology_url: string | null;
+  datasource_id: string;
+}
+
+interface EmissionsFactorValue {
+  id: string;
+  gas: string;
+  gpc_reference_number: string;
+  scope: string;
+  units: string;
+  ar4_gwp: number | null;
+  ar5_gwp: number | null;
+  ar6_gwp: number | null;
+  co2_equivalent_value: number;
+  co2_equivalent_units: string;
+  activity_value: number;
+  activity_units: string;
+  emissions_factor_value: number;
+  year: number | null;
+  region: string | null;
+  source: string;
+  reference: string;
+  description: string;
+  temporal_granularity: string;
+  url: string;
+  methodology_id: string;
+  metadata: string;
+  deprecated?: boolean;
+}
+
+interface EmissionsFactorDataSourceMapping {
+  datasource_id: string;
+  emissions_factor_id: string;
+}
+
+interface APIResponse<T> {
+  [key: string]: T[];
+}
+
+async function fetchEmissionsFactorData(baseUrl: string) {
+  console.log("Fetching emissions factor data from global API...");
+  
+  const endpoints = [
+    { path: "/emissions_factor/publisher", key: "emissions_factor_publisher" },
+    { path: "/emissions_factor/datasource", key: "emissions_factor_datasource" },
+    { path: "/emissions_factor/methodology", key: "emissions_factor_methodologies" },
+    { path: "/emissions_factor/emissions_factor", key: "emissions_factor" },
+    { path: "/emissions_factor/emissions_factor_datasource", key: "emissions_factor_datasource_mapping" }
+  ];
+
+  const results: Record<string, any> = {};
+
+  for (const endpoint of endpoints) {
+    const url = `${baseUrl}${endpoint.path}`;
+    console.log(`Fetching ${url}`);
+    
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error(
+        `Failed to fetch ${endpoint.path} with error ${response.status} ${response.statusText}`
+      );
+    }
+    
+    const data = await response.json();
+    // Handle the datasource mapping endpoint which returns emissions_factor_datasource
+    if (endpoint.path === "/emissions_factor/emissions_factor_datasource") {
+      results[endpoint.key] = data["emissions_factor_datasource"] || [];
+    } else {
+      results[endpoint.key] = data[endpoint.key] || [];
```

Comment:
Can we try to avoid this special casing? Why can't the key be used to index the EF data source endpoint?
If there's an issue with the consistency of the global API result schemas, let's bring this up as an issue directly 👍